packages/@react-spectrum/combobox/src/MobileComboBox.tsx

```diff
@@ -477,7 +478,7 @@ function ComboBoxTray(props: ComboBoxTrayProps) {
           onScroll={onScroll}
           onLoadMore={onLoadMore}
           isLoading={loadingState === 'loading' || loadingState === 'loadingMore'} />
-        <DismissButton onDismiss={() => state.commit()} />
+        <DismissButton onDismiss={chain(state.commit, state.close)} />
```

Comment:
@LFDanLu since u are online i have a question regarding `chain`.(might be dumb) 

Might we have situations when `close` should be called after `commit` finishes, what i mean will we encounter situations when `commit` might be `async` resulting in `close` removing a `component` from the screen and `commit` did not finish ?

Reply:
Hm, I can't think of a scenario where this would be an issue. [`commit`](https://github.com/adobe/react-spectrum/blob/main/packages/@react-stately/combobox/src/useComboBoxState.ts#L199-L212) updates the input value/selected key, don't think `close` would interfere with it

Reply:
i was thinking on a situation where `commit` does some `setState` and then `close` comes in and `unmounts` the `component`,  should get an error in that situation

Reply:
The tray would be unmounted, but the MobileCombobox itself would remain.

Reply:
👍